R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+#' comparing the absolute numbers of the age group 20 to 24. This has the
+#' additional advantage that for this age group anthropological aging
+#' methods are reasonable exact.
+#'
+#' @param life_table an object of class mortaar_life_table_list.
+#' @param females an object of class mortaar_life_table for females
+#' @param males an object of class mortaar_life_table for males
+#'
+#' @return Output of masculinity index and maternal mortality.
+#' \itemize{
+#'   \item \bold{Masculinity index}.
+#'
+#'                    \eqn{MI = D>=15male * 100 / D>=15female}
+#'
+#'   \item \bold{Maternal mortality}.
+#'
+#'                    \eqn{333.33 * D20-24female / D20-24male - 76.07}
+#'}
+#'
+#' @examples
+#' # Calculate Masculinity index and maternal mortality from real life
+#' # dataset.
+#' library(magrittr)
+#' muen <- muensingen
+#' muen %>% dplyr::select(age) %>% dplyr::arrange(nchar(age), age) %>%
+#' unique()
+#' muen <- muen %>% dplyr::mutate( age = dplyr::case_when(
+#' .$age == ">50"    ~ "50-70", .$age == ">60"    ~ "60-70",
+#' .$age == "10+/-1" ~ "9-12", .$age == "10"     ~ "10-11",
+#' .$age == "7-8"    ~ "7-9", .$age == "9-10"   ~ "9-11",
+#' TRUE ~  .$age))
+#' muen <- muen %>% tidyr::separate(age, c("from", "to")) %>%
+#' transform(from = as.numeric(from), to = as.numeric(to))
+#' muen_prep <- muen %>% prep.life.table(group = "sex",
+#' agebeg = "from", ageend = "to", method = "Standard",
+#' agerange = "excluded")
+#' muen_result <- muen_prep %>% life.table()
+#' lt.sexrelation(muen_result,"female","male")
+#'
+#' @export
+#'
+lt.sexrelation <- function(life_table, females, males) {
+
+  # check if life_table is of class "mortaar_life_table"
+  if (!inherits(life_table,"mortaar_life_table_list"))  {
+    paste0(
+      "The dataset does not seem to be a mortAAR life table list."
+    ) %>%
+      message
+  } else {
+
+    # Masculinity index (MI) for juvenile and older individuals: males * 100 / females
+
+    fem_age <- life_table[females][[1]][[2]] %>% cumsum
+    fem_sum <- life_table[females][[1]][[4]][which(fem_age >= 20)] %>% sum
+    male_age <- life_table[males][[1]][[2]] %>% cumsum
+    male_sum <- life_table[males][[1]][[4]][which(male_age >= 20)] %>% sum
+
+    masculinity_index <- round(male_sum * 100 / fem_sum, 1)
+
+
+    # compute maternal mortality according to McFaden/Oxenham 2019
+    f_mort <- life_table[females][[1]][[4]][which(life_table[females][[1]][[1]]=="20--24")]
+    m_mort <- life_table[males][[1]][[4]][which(life_table[males][[1]][[1]]=="20--24")]
+    maternal_mortality <- (333.33 * f_mort / m_mort) - 76.07
+    maternal_mortality_rate <- round(100000 / maternal_mortality)
+
+    cat(sep='', "Masculinity index: \n", masculinity_index,
+        "\n\nMaternal mortality:\n", round(maternal_mortality,1),
+        "\n\nOne of ", maternal_mortality_rate, " women is dying during",
+        " pregnancy or within the first 42 days after birth due to",
+        " complications.")
+  }
```

Comment:
1. Please avoid `cat` everywhere. You should use `message` for console messages.
2. No function should only print a message. The values should also be numerically accessible. `lt.sexrelation` should return a list with the two output values.

Reply:
It now returns a data.frame.